server/src/main/java/io/druid/segment/indexing/RealtimeTuningConfig.java

```diff
@@ -55,7 +55,14 @@
 
   private static File createNewBasePersistDirectory()
   {
-    return Files.createTempDir();
+    try {
+      return Files.createTempDir();
+    }
+    catch (IllegalStateException e) {
+      String message = "Failed to create temporary directory!" + System.lineSeparator() +
```

Comment:
Nit: error messages with line breaks can make it tougher to grep through logs, and also can look weird, so I would leave it out.

Reply:
I would also include the current `java.io.tmpdir` property here, something like, "Failed to create temporary directory in[%s]" & format that with `java.io.tmpdir`. Using `ISE` instead of `IllegalStateException` can help here (it does string formatting in the constructor).

Reply:
Thanks, fixed